twilight-gateway/src/shard.rs

```diff
@@ -781,6 +770,13 @@ impl Shard {
             .cloned()
             .unwrap_or_else(default_identify_properties);
 
+        tracing::debug!(shard_id = %self.id(), "queued for identify");
+        self.config
+            .queue()
+            .request([self.id.number(), self.id.total()])
+            .await;
```

Comment:
This blocks execution and should probably be, like in #1947, moved inside `NextMessageFuture`